### ISSUE TITLE
Error on comparing response from model function to boolean

### ISSUE THREAD

**Issue by PiousVenom** [nunomaduro/larastan#832]
- Larastan Version: 0.7.6
- `--level` used: 5
- Pull request with failing test: 

### Description
Getting error ` Strict comparison using === between Illuminate\Database\Eloquent\Builder<App\Models\Publisher> and false will always evaluate to false.` when comparing the response of calling `restore()` on a model.

### Laravel code where the issue was found

```php
$publisher = Publisher::withTrashed()->where('id', $id);
$response = $publisher->restore();

if ($response === false)
{
}
```

**Comment by canvural** [comment]
Fixed with e02c9750f82800c05592493efc9c3acb259f753a

**Comment by szepeviktor** [comment]
# 🏎️
